src/libarena/lib.rs

```diff
@@ -632,6 +638,96 @@ mod tests {
         }
     }
 
+    #[test]
+    pub fn test_typed_arena_zero_sized() {
+        let arena = TypedArena::new();
+        for _ in 0..100000 {
+            arena.alloc(());
+        }
+    }
+
+    #[test]
+    pub fn test_arena_zero_sized() {
+        let arena = Arena::new();
+        for _ in 0..1000 {
+            for _ in 0..100 {
+                arena.alloc(|| ());
+            }
+            arena.alloc(|| Point { x: 1, y: 2, z: 3 });
```

Comment:
Is there something meaningful we can test here? Maybe that the values aren't clobbering each other?